### PR TITLE
Add print cpu feature info port library API

### PR DESCRIPTION
**Pull request by harryyu1994**

### CODE REVIEW

**Comment by harryyu1994** [comment]
@fjeremic Hi Filip, could you help merge this one?

File: port/common/omrport.c [hunk 1]
```diff
@@ -76,6 +76,7 @@ static OMRPortLibrary MainPortLibraryTable = {
 	omrsysinfo_processor_has_feature, /* sysinfo_processor_has_feature */
 	omrsysinfo_processor_set_feature, /* sysinfo_processor_set_feature */
 	omrsysinfo_get_processor_feature_name, /* sysinfo_get_processor_feature_name */
+	omrsysinfo_print_processor_features, /* sysinfo_print_processor_features */
```
Comment (fjeremic) [inline port/common/omrport.c:79]: Looking at the other APIs that return strings I don't see any that use "print" in the name. Perhaps `omrsysinfo_get_processor_feature_string` makes more sense here?

File: port/common/omrsysinfo.c [hunk 2]
```diff
@@ -163,6 +163,40 @@ omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32
 	return rc;
 }
 
+/**
+ * Generate the corresponding string literals for the provided OMRProcessorDesc.
+ *
+ * @param[in] portLibrary The port library.
+ * @param[out] buff The cpu features output string.
+ * @param[in] BUFF_SIZE The size of the buff.
```
Comment (fjeremic) [inline port/common/omrsysinfo.c:171]: You should mention that the buffer will be zero initialized and overwritten with the string contents. The size argument should also say "in number of bytes".

File: port/common/omrsysinfo.c [hunk 3]
```diff
@@ -163,6 +163,40 @@ omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32
...
+ */
+void
+omrsysinfo_print_processor_features(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc, char * buff, const size_t BUFF_SIZE)
```
Comment (fjeremic) [inline port/common/omrsysinfo.c:174]: Looking at omrport.h many of the similar type APIs name the two arguments `buffer` and `length`. I would say we follow suit and have some consistency here.

File: port/common/omrsysinfo.c [hunk 4]
```diff
@@ -163,6 +163,40 @@ omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32
...
+					strncat(buff, " ", BUFF_SIZE - strlen(buff) - 1);
+					rowLength += 1;
+				}
+				strncat(buff, feature_name, BUFF_SIZE - strlen(buff) - 1);
+				rowLength += strlen(feature_name);
+		    }
+		}
+	}
+	strncat(buff, "\n", BUFF_SIZE - strlen(buff) - 1);
+	return;
```
Comment (fjeremic) [inline port/common/omrsysinfo.c:197]: The `return;` statement seems redundant as there are other examples of `void` functions in this file not using it.

File: port/common/omrsysinfo.c [hunk 5]
```diff
@@ -163,6 +163,40 @@ omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32
...
+void
+omrsysinfo_print_processor_features(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc, char * buff, const size_t BUFF_SIZE)
+{
+	memset(buff, 0, BUFF_SIZE*sizeof(char));
+	size_t rowLength = 0;
+	for (size_t i = 0; i < OMRPORT_SYSINFO_FEATURES_SIZE; i++) {
+		size_t numberOfBits = CHAR_BIT * sizeof(desc->features[i]);
+		for (size_t j = 0; j < numberOfBits; j++) {
+
+			if (desc->features[i] & (1<<j)) {
```
Comment (fjeremic) [inline port/common/omrsysinfo.c:182]: Spaces around binary operators needed. Similarly elsewhere (`*sizeof` above, etc.).

File: port/common/omrsysinfo.c [hunk 6]
```diff
@@ -163,6 +163,40 @@ omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32
...
+ *
+ * @param[in] portLibrary The port library.
+ * @param[out] buff The cpu features output string.
+ * @param[in] BUFF_SIZE The size of the buff.
+ */
+void
+omrsysinfo_print_processor_features(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc, char * buff, const size_t BUFF_SIZE)
+{
+	memset(buff, 0, BUFF_SIZE*sizeof(char));
+	size_t rowLength = 0;
```
Comment (fjeremic) [inline port/common/omrsysinfo.c:177]: This variable name is misleading. It seems to be used as a write cursor and should be named accordingly. There are also no rows as it is all written in one line. I think we should re-adapt the name.

File: port/unix/omrsysinfo.c [hunk 7]
```diff
@@ -710,6 +710,40 @@ omrsysinfo_get_processor_feature_name(struct OMRPortLibrary *portLibrary, uint32
...
 
+/**
+ * Generate the corresponding string literals for the provided OMRProcessorDesc.
+ *
+ * @param[in] portLibrary The port library.
+ * @param[out] buff The cpu features output string.
+ * @param[in] BUFF_SIZE The size of the buff.
+ */
+void
+omrsysinfo_print_processor_features(struct OMRPortLibrary *portLibrary, OMRProcessorDesc *desc, char * buff, const size_t BUFF_SIZE)
```
Comment (fjeremic) [inline port/unix/omrsysinfo.c:721]: This looks like an identical copy. Any chance we can get away without this code duplication?

**Review by fjeremic — CHANGES_REQUESTED** [review]

**Comment by harryyu1994** [comment]
Ready for review again.

File: port/unix/omrsysinfo.c (hunk 7 above)
Comment (harryyu1994) [reply port/unix/omrsysinfo.c]: I looked like how other APIs are implemented. 
I think we have to duplicate it for windows and unix but we can remove the implementation for common.
It seems to me that the common file (`port/common/omrsysinfo.c`) is like an interface file where we don't add the implementation.

**Review by fjeremic — APPROVED** [review]

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by fjeremic** [comment]
@harryyu1994 see failures.

**Comment by harryyu1994** [comment]
Please try again.
```cpp
15:14:54  /home/jenkins/workspace/Build/port/unix/omrsysinfo.c:727:2: error: ‘for’ loop initial declarations are only allowed in C99 mode
15:14:54    for (size_t i = 0; i < OMRPORT_SYSINFO_FEATURES_SIZE; i++) {
15:14:54    ^
15:14:54  /home/jenkins/workspace/Build/port/unix/omrsysinfo.c:727:2: note: use option -std=c99 or -std=gnu99 to compile your code
15:14:54  /home/jenkins/workspace/Build/port/unix/omrsysinfo.c:729:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
15:14:54     for (size_t j = 0; j < numberOfBits; j++) {
15:14:54     ^
```

**Comment by fjeremic** [comment]
@genie-omr build all

**Review by fjeremic — APPROVED** [review]

**Pull request merged**